File: javascript/rna-transcription/rna-transcription.js

var DnaTranscriber = function () {};

DnaTranscriber.prototype.toRna = function (sequence) {
  let mapping = {G: "C", C: "G", T: "A", A: "U"};
  var rnaSequence = "";
  let splitSequence = sequence.split("");
  for (var base of splitSequence) {
    if (["G", "C", "A", "T"].includes(base))
      rnaSequence += mapping[base];
    else
      throw new Error("Invalid input");
  }
  return rnaSequence;
};

module.exports = DnaTranscriber;

File: python/rna-transcription/rna_transcription.py
def to_rna(dna_strand):
    rna = []
    transpose = {"A": "U", "T": "A", "C": "G", "G": "C"}
    for base in list(dna_strand):
        if base not in ["A", "T", "C", "G"]:
            return ""
        rna.append(transpose[base])
    return "".join(rna)
        
